pliers/extractors/text.py

```
@@ -453,98 +446,282 @@ def __init__(self,
+            (valid values include all the models with 'bert' prefix) 
+            or path to custom model.
+        tokenizer (str): Type of tokenization used in the tokenization step.
+            If different from model, out-of-vocabulary tokens may be treated as
+            unknown tokens.
+        framework (str): name deep learning framework to use. Must be 'pt'
+            (PyTorch) or 'tf' (tensorflow). Defaults to 'pt'.
+        pooling (str): defines whether to return encoding for [CLS] token 
+            (None, default), or the numpy function to use to pool token-level 
+            encodings.
+        return_sep (bool): defines whether to return encoding for the [SEP]
+            token.
+        return_metadata (bool): If True, the extractor returns an additional 
+            feature column with the encoded sequence.
+        model_kwargs (dict): Named arguments for pretrained model.
+            See: https://huggingface.co/transformers/main_classes/model.html
+            and https://huggingface.co/transformers/model_doc/bert.html
+        tokenizer_kwargs (dict): Named arguments for tokenizer.
+            See https://huggingface.co/transformers/main_classes/tokenizer.html
+    '''
+
+    _log_attributes = ('pretrained_model', 'framework', 'tokenizer_type', 
+                       'pooling', 'return_sep', 'model_class', 'model_kwargs',
+                       'tokenizer_kwargs')
+
+    def __init__(self,
+                 pretrained_model='bert-base-uncased',
+                 tokenizer='bert-base-uncased',
+                 framework='pt',
+                 pooling=None,
+                 return_sep=False,
+                 return_metadata=False,
+                 model_kwargs=None,
+                 tokenizer_kwargs=None):
+
+        super(BertSequenceEncodingExtractor, self).__init__(pretrained_model,
+            tokenizer, framework, return_metadata, model_kwargs, 
+            tokenizer_kwargs, model_class='BertModel')
+        if pooling:
```

Would probably move this check above the superclass initializer to fail faster.